### PR TITLE
feat(#11): load data from users (fixing the error from the last branch)

### PR DESCRIPTION
**Pull request by snooopyman**

### CODE REVIEW

File: ProgrammaticUI.xcodeproj/project.xcworkspace/xcuserdata/armando.caceres.xcuserdatad/UserInterfaceState.xcuserstate
Comment (MartinSandeCosta) [inline ProgrammaticUI.xcodeproj/project.xcworkspace/xcuserdata/armando.caceres.xcuserdatad/UserInterfaceState.xcuserstate:1]: Consider including this in your gitignore

Comment (MartinSandeCosta) [reply ProgrammaticUI.xcodeproj/project.xcworkspace/xcuserdata/armando.caceres.xcuserdatad/UserInterfaceState.xcuserstate:1]: The same for `Package.resolved`

File: ProgrammaticUI/Controllers/HomeViewController.swift [hunk 1]
```diff
@@ -65,6 +67,39 @@ class HomeViewController: UIViewController {
         navigationController?.pushViewController(insertDataVC, animated: true)
     }
 
+    private func loadData() {
+        guard let userEmail = Auth.auth().currentUser?.email else {
+            fatalError("User not authenticated.")
```
Comment (MartinSandeCosta) [inline ProgrammaticUI/Controllers/HomeViewController.swift:72]: This stops execution. Do we want to crash if `Auth.auth().currentUser?.email` is nil?

File: ProgrammaticUI/Controllers/HomeViewController.swift [hunk 2]
```diff
@@ -65,6 +67,39 @@ class HomeViewController: UIViewController {
...
+        }
+
+        let ref = Database.database().reference().child("users").child(getUsername(from: userEmail))
```
Comment (MartinSandeCosta) [inline ProgrammaticUI/Controllers/HomeViewController.swift:75]: I would statically allocate `users`, `km`, `calories` and `duration` as properties in a new struct so they can be commonly accessed here and in `InserDataVC`.

File: ProgrammaticUI/Controllers/HomeViewController.swift (hunk 1 above)
Comment (MartinSandeCosta) [reply ProgrammaticUI/Controllers/HomeViewController.swift:72]: Same in `getUsername(from:)`

**Review by MartinSandeCosta — CHANGES_REQUESTED** [review]

File: ProgrammaticUI/Controllers/HomeViewController.swift (hunk 1 above)
Comment (snooopyman) [reply ProgrammaticUI/Controllers/HomeViewController.swift:72]: Realmente nunca llega a ejecutarse eso porque primero se valida el usuario. No deja saltar a la siguiente pantalla si aparece un login raro. 
Lo acabo de probar y no se rompe. lo único que salta por consola es: "The email address is badly formatted."
Si pongo ambos vacíos: "The password is invalid or the user does not have a password."

Entiendo que está mal hecho de esta manera pero en qué momento podría romperse sabiendo esto ? y cuál sería la mejor manera?  solo usando un if let y lanzando un print o incluso con el guard pero lanzando un UIAlert mostrando el error no ?

Comment (MartinSandeCosta) [reply ProgrammaticUI/Controllers/HomeViewController.swift:72]: In a production app, you could present the user with some visual feedback like an alert window. Also, I would log this in two different ways:

1. When building for production; log the failure event in Firebase and present the user with a warning.
2. When building for debugging; log the failure event locally.

Comment (MartinSandeCosta) [reply ProgrammaticUI/Controllers/HomeViewController.swift:72]: On the other topic you mentioned and to answer your question: this stops execution when `Auth.auth().currentUser` is `nil`.

**Pull request merged**